pkg/apis/pipeline/v1alpha1/pipeline_validation.go

```diff
@@ -237,6 +319,16 @@ func validatePipelineWorkspaces(wss []WorkspacePipelineDeclaration, pts []Pipeli
 			}
 		}
 	}
+	for tIdx, t := range finalTasks {
+		for wsIdx, ws := range t.Workspaces {
```

Comment:
do we really need to use these indexes? maybe we could ignore them with _? not sure how much the adde to the error messages

Reply:
yup, index is used to notify which workspace in the `PipelineTask` is not declared in Pipeline spec in case its missing from Pipeline's workspace list.